Blogpost for announcing Kubernetes Upstream Community Group Annual Reports (#28435)
kubernetes/website: content/en/blog/_posts/2021-06-18-Announcing-Kubernetes-Community-Group-Annual-Reports.md

  Special Interest Groups and Working Groups.
date: 2021-06-23T10:00:00-08:00
slug: Announcing-Kubernetes-Community-Group-Annual-Reports
---

**Authors:** Divya Mohan

Given the growth and scale of the Kubernetes project, the existing reporting mechanisms were proving to be inadequate and challenging.
Kubernetes is a large open source project. With over 100000 commits just to the main k/kubernetes repository, hundreds of other code
repositories in the project, and thousands of contributors, there's a lot going on. In fact, there are 37 contributor groups at the time of
writing. We also value all forms of contribution and not just code changes.

With that context in mind, the challenge of reporting on all this activity was a call to action for exploring better options. Therefore
inspired by the Apache Software Foundation’s [open guide to PMC Reporting](https://www.apache.org/foundation/board/reporting) and the
[CNCF project Annual Reporting](https://www.cncf.io/cncf-annual-report-2020/), the Kubernetes project is proud to announce the 
**Kubernetes Community Group Annual Reports for Special Interest Groups (SIGs) and Working Groups (WGs)**. In its flagship edition, 

> the [2020 Summary report](https://github.com/kubernetes/steering/blob/master/reports/summary-2020.md) focuses on bettering the 
> Kubernetes ecosystem by assessing and promoting the healthiness of the groups within the upstream community. 

Reviewer: ```suggestion
the [2020 Summary report](https://www.cncf.io/kubernetes-community-annual-report-2020/) focuses on bettering 
the Kubernetes project by assessing and promoting the healthiness of the groups within the community. 
```

switched to CNCF report
cc @parispittman do you think we should switch it or keep it pointed at the GH report?

Author: @parispittman : Will await your response on this. I have committed the other suggestions made by y'all :)

Reviewer 2: I prefer linking to https://www.cncf.io/kubernetes-community-annual-report-2020/

Reviewer 3: +1 to @sftim

Previously, the mechanisms for the Kubernetes project overall to report on groups and their activitie were 

> (devstats)[https://k8s.devstats.cncf.io/], GitHub data, issues, to measure the healthiness of a given UG/WG/SIG/Committee. As a 
> project spanning several diverse communities, it was essential to have something that captured the human side of things. With 50,000+
> contributors, it’s easy to assume that the project has enough help and this report surfaces more information than /help-wanted and 
> /good-first-issue for end users. This is how we sustain the project. Paraphrasing one of the Steering Committee members, 

Reviewer: ```suggestion
[devstats](https://k8s.devstats.cncf.io/), GitHub data, issues, and their quarterly Community Meeting readout to gauge the healthiness of a given Community Group. As a 
project spanning several diverse communities, it was essential to have something that captured the human side of things. With 50,000+
contributors, it’s easy to assume that the project has enough help, but that is far from the case, with the report
surfacing more information than what can be found via [`/help-wanted`](http://go.k8s.io/help-wanted)
and [`/good-first-issue`](http://go.k8s.io/good-first-issue) labeled issued.
This is how we sustain the project. Paraphrasing one of the Steering Committee members, 
```
We only mention SIG/WG earlier, switching to community groups covers everything. Also linked to our help wanted/gfi links in case folks do want to go that route heh

[Paris Pittman](https://github.com/parispittman), “There was a requirement for tighter feedback loops - ones that involved more than just
GitHub data and issues. Given that Kubernetes, as a project, has grown in scale and number of contributors over the years, we have 
outgrown the existing reporting mechanisms."

The existing communication channels between the Steering committee members and the folks leading the groups and committees were also required

> to be made as open and as bi-directional as possible. Towards achieving this very purpose, every group and committee has been assigned a

Reviewer: ```suggestion
to be made as open and as bi-directional as possible. Towards achieving this purpose, every group and committee has been assigned a
```
small nit

liaison from among the steering committee members for kick off, help, or guidance needed throughout the process. According to 
[Davanum Srinivas a.k.a. dims](https://github.com/dims), “... That was one of the main motivations behind this report. People (leading the
groups/committees) know that they can reach out to us and there’s a vehicle for them to reach out to us… This is our way of setting up a

> two-way feedback for them." The progress on these action items would be updated and tracked on the monthly Steering Committee meetings
> ensuring that this is not a one-off activity. Quoting [Nikhita Raghunath](https://github.com/nikhita), one of the Steering Committee members,
> “... Once we have a base, the liaisons will work with these groups to ensure that the problems are resolved. When we have a report next year, 
> we’ll have a look at the progress made and how we could still do better. But the idea is definitely to not stop at the report.”

Reviewer 2: I'd add a paragraph break:

```suggestion
two-way feedback for them."

The progress on these action items would be updated and tracked on the monthly Steering Committee meetings
ensuring that this is not a one-off activity. Quoting [Nikhita Raghunath](https://github.com/nikhita), one of the Steering Committee members,
“… Once we have a base, the liaisons will work with these groups to ensure that the problems are resolved. When we have a report next year, 
we’ll have a look at the progress made and how we could still do better. But the idea is definitely to not stop at the report.”
```

description: >
  Introducing brand new Kubernetes Community Group Annual Reports for
  Special Interest Groups and Working Groups.
date: 2021-06-23T10:00:00-08:00
slug: Announcing-Kubernetes-Community-Group-Annual-Reports
---

**Authors:** Divya Mohan

Given the growth and scale of the Kubernetes project, the existing reporting mechanisms, such as the quarterly read-outs at the Kubernetes Community Meeting, were proving to be inadequate and challenging.
Kubernetes is a large open-source project. With over 100000 commits just to the main [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) repository, hundreds of other
repositories, 37 contributor groups, and thousands of contributors, both code and non-code alike, there's a lot going on.

With that context in mind, the challenge of reporting on all this activity was a call to action for exploring better options. Therefore
inspired by the Apache Software Foundation’s [open guide to PMC Reporting](https://www.apache.org/foundation/board/reporting) and the
[CNCF project Annual Reporting](https://www.cncf.io/cncf-annual-report-2020/), the Kubernetes project is proud to announce the 

> **Kubernetes Community Group Annual Reports for Special Interest Groups (SIGs) and Working Groups (WGs)**. In its flagship edition, 

Reviewer 2: Is this better:
```suggestion
**Kubernetes Community Group Annual Reports for Special Interest Groups (SIGs) and Working Groups (WGs)**. In its inaugural edition,
```
?
